test/distributed/org/apache/cassandra/distributed/test/cql3/SingleNodeTableWalkTest.java

```diff
@@ -363,7 +386,8 @@ protected TableMetadata defineTable(RandomSource rs, String ks)
                                        .withCompression())
                      .withKeyspaceName(ks).withTableName("tbl")
                      .withSimpleColumnNames()
-                     .withDefaultTypeGen(supportedTypes())
+                     .withDefaultTypeGen(supportedTypes(rs))
+                     .withPrimaryColumnTypeGen(supportedPrimaryColumnTypes(rs))
```

Comment:
when primary keys can be collections there can be issues with having too many bytes and having our mutations be rejected.  Keeping it primitive also improves readability for debug output...

Would be good to have a test that overrides this though, just to improve coverage...